lib/pushmi_pullyu/aip/solr_fetcher.rb

```diff
@@ -1,47 +1,33 @@
-require 'pushmi_pullyu'
-require 'net/http'
 require 'json'
+require 'net/http'
 
 class PushmiPullyu::AIP::SolrFetcher
 
+  class SolrFetchError < StandardError; end
+
   def initialize(noid)
     @noid = noid
   end
 
   def fetch_permission_object_ids
-    json = run_query_json("accessTo_ssim:#{@noid}", fields: 'id', url_extra: nil)
-    hash = JSON.parse(json)
+    hash = JSON.parse(run_query_json)
+
     return [] if hash['response']['docs'].empty?
+
     hash['response']['docs'].map { |hit| hit['id'] }
   end
 
   private
 
-  def solr_url
-    PushmiPullyu.options[:solr][:url]
-  end
-
-  def query_url
-    "#{solr_url}/select"
-  end
-
-  def run_query_json(query, fields: nil, url_extra: nil)
-    # Return fetched results, else raise an error
-    url = "#{query_url}?q=#{query}"
-    url += "&fl=#{fields}" if fields
-    url += '&wt=json'
-    url += url_extra if url_extra
-    uri = URI(url)
-
-    request = Net::HTTP::Get.new(uri)
-
-    response = Net::HTTP.start(uri.hostname, uri.port) do |http|
-      http.request(request)
-    end
+  # Return fetched results, else raise an error
+  def run_query_json
```

Comment:
Reduced majority of this code down to 3 lines. Much easier to read and understand now as a bonus.

Using Net:HTTP build in `get_response` method as well since were just doing a simple get request for the response and nothing else.